_test/test_utilities_herbert/test_str_to_iarray.m

```diff
@@ -0,0 +1,187 @@
+classdef test_str_to_iarray < TestCase
+    % Test test_str_to_iarray
+    %
+    % str_to_iarray converts a string, character array or cell array of strings
+    % to an integer array
+    
+    methods
+        %--------------------------------------------------------------------------
+        % Single number
+        %--------------------------------------------------------------------------
+        function test_single_number (~)
+            % Test constructor with no arguments
+            iarr = str_to_iarray ('37');
+            assertEqual (iarr, 37)
+        end
+        
+        %--------------------------------------------------------------------------
+        % Matlab style tokens
+        %--------------------------------------------------------------------------
+        function test_range (~)
+            iarr = str_to_iarray ('-3:6');
+            assertEqual (iarr, -3:6)
+        end
+        
+        function test_range_rhs_lt_lhs (~)
+            iarr = str_to_iarray ('6:-3');
+            assertEqual (iarr, 6:-3)
```

Comment:
hides the underlying value which is empty double size 1x0.  assert should check for this latter

Reply:
I appreciate that, but what I was hoping to make explicit is that the parsing is correct, rather than what Matlab has chosen to do. For that reason I prefer to leave as is.

Reply:
@tgperring this does not say that the parsing is correct. It just says that iarr has the value empty double 1x0. It says it in as obscure and roundabout a way as possible. The test does not show that 6:-3 was processed correctly, only that it produced the same, final, ambiguous result.

Reply:
On reflection, I agree, as there is a difference between e.g. [] and zeros(1,0) as the output and it is specifically the latter  which str_to_iarray needs to produce. I've made the requested change to this and also the other tests that are supposed to produce zeros(1,0).